test/L2/L2Reward.t.sol

```diff
@@ -210,141 +219,154 @@ contract L2RewardTest is Test {
+        l2Reward.increaseLockingAmount(increasingAmounts);
+        vm.stopPrank();
+    }
 
-        checkConsistencyPendingUnlockDailyUnlocked(lockIDs, getLargestExpiryDate(lockIDs));
+    function stakerClaimRewards(uint256 stakerIndex, Scenario memory scenario) private {
+        uint256[] memory positionsToBeModified = new uint256[](1);
+        positionsToBeModified[0] = scenario.lockIDs[stakerIndex];
+        vm.prank(scenario.stakers[stakerIndex]);
+        l2Reward.claimRewards(positionsToBeModified);
+    }
 
-        vm.warp(19740 days + 50 days);
-        // extend stake 1 by 30 days
-        L2Reward.ExtendedDuration[] memory durationExtensions = new L2Reward.ExtendedDuration[](1);
-        durationExtensions[0].lockID = lockIDs[0];
-        durationExtensions[0].durationExtension = 30;
-        vm.prank(stakers[0]);
-        l2Reward.extendDuration(durationExtensions);
+    function allStakersClaim(Scenario memory scenario) private {
+        uint256[] memory positionsToBeModified = new uint256[](1);
+        scenario.lastClaimedAmount = 0;
+        for (uint8 i = 0; i < scenario.stakers.length; i++) {
+            if (scenario.lockIDs[i] == 0) {
+                continue;
+            }
 
-        vm.warp(19740 days + 70 days);
-        // stake 6: (100 LSK, 150 days)
-        // Increases amount at day 100 by 50
-        lockIDs[5] = when_stakerCreatesPosition(
-            stakers[5], Position({ amount: convertLiskToSmallestDenomination(100), duration: 150 })
-        );
+            positionsToBeModified[0] = scenario.lockIDs[i];
+            vm.prank(scenario.stakers[i]);
+            l2Reward.claimRewards(positionsToBeModified);
 
-        checkConsistencyPendingUnlockDailyUnlocked(lockIDs, getLargestExpiryDate(lockIDs));
+            scenario.balances[i] = l2LiskToken.balanceOf(scenario.stakers[i]);
 
-        vm.warp(19740 days + 80 days);
-        // resumes stake 2 on day 80
-        positionsToBeModified[0] = lockIDs[1];
-        vm.prank(stakers[1]);
-        l2Reward.resumeUnlockingCountdown(positionsToBeModified);
+            scenario.lastClaimedAmount += scenario.balances[i] - scenario.oldBalances[i];
+        }
+    }
 
-        vm.warp(19740 days + 89 days);
-        // Pauses stake 5 on day 89
-        positionsToBeModified[0] = lockIDs[4];
-        vm.prank(stakers[4]);
-        l2Reward.pauseUnlocking(positionsToBeModified);
+    function cacheBalances(Scenario memory scenario) private view {
+        for (uint8 i = 0; i < scenario.stakers.length; i++) {
+            scenario.oldBalances[i] = l2LiskToken.balanceOf(scenario.stakers[i]);
+        }
+    }
 
-        vm.warp(19740 days + 95 days);
-        // stake 7: (200 LSK, 200 days)
-        lockIDs[6] = when_stakerCreatesPosition(
-            stakers[6], Position({ amount: convertLiskToSmallestDenomination(200), duration: 200 })
-        );
+    function checkRewardsContractBalance(uint256 funds) public {
```

Comment:
I see this function is only called in scenarios 1 and 4. Should we call it in other scenarios as well for better coverage?

Reply:
Also, to become easier to parse, I would suggest adding a comment before the function definition explaining what is the function doing and that it should always be called after all stakers claimed their rewards (otherwise the check of line 337 would not be true..)

Reply:
scenario2 and scenario3 deletes locking position so doesn't fit for assertions in checkRewardsContract.

Reply:
This is modified all scenarios check rewards contact balance.